cpp/csp/adapters/CMakeLists.txt

```diff
@@ -8,7 +8,10 @@ if(CSP_BUILD_PARQUET_ADAPTER)
 endif()
 
 if(CSP_BUILD_WS_CLIENT_ADAPTER)
-    add_subdirectory(websocket)
+    # Need to build websocket adapter under cpp17 standard due to websocketpp incompatibility issues
+    set(CMAKE_CXX_STANDARD 17)
```

Comment:
I think you can set this in the websocket cmake it self ( which would make more sense ) and then you shouldnt have to set it back either